src/mods.cpp

```diff
@@ -158,8 +173,8 @@ ModConfiguration::ModConfiguration(std::string worldpath)
 	SubgameSpec gamespec = findWorldSubgame(worldpath);
 
 	// Add all game mods and all world mods
-	addModsInPath(gamespec.gamemods_path);
-	addModsInPath(worldpath + DIR_DELIM + "worldmods");
+	addModsInPath(gamespec.gamemods_path, worldpath);
+	addModsInPath(worldpath + DIR_DELIM + "worldmods", worldpath);
```

Comment:
Passing the worldpath means that the configuration is loaded twice here.  Can you load it once and just pass a `const Settings &`?